### PR TITLE
Report generation

### PR DESCRIPTION
**Pull request by psahni**
- LOP & LOP refund report  generation
- Option to choose fields - employee code, manager, start_date, end_date
- check  box to select all employees report.
- task to schedule attendance
- task:- `rake attendance`
- Association modification. User can have multiple projects. https://github.com/psahni/trantor_management_system/commit/fcdb9de89f1d2a56537a0b979803218157a8ed37

### CODE REVIEW

**Comment by psahni** [comment]
@ramanbedi1989 - Please if you can review the PR.
Thanks

File: app/controllers/admin/reports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		@loss_of_pays = LossOfPay.includes(:loss_of_pay_infos)
+  end
+
+  def create
+    begin
+      @report = Report.new(params)
+      unless @report.valid?
+        return render action: :loss_of_pays
+      end
+      case params[:report_type].downcase
```
Comment (ramanbedi1989) [inline app/controllers/admin/reports_controller.rb]: When you have used downcase for the report_type, I don't think it'll ever match "when 'lopRefund'" case as 'lopRefund' is in camelCase.

File: app/controllers/admin/reports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        when 'lop'
+          file = @report.generate_lop_report
+          send_file(file)
+        when 'lopRefund'
+          @report.print_lop_refunt
```
Comment (ramanbedi1989) [inline app/controllers/admin/reports_controller.rb]: couldn't find print_lop_refunt function in Report Model

Comment (psahni) [inline app/controllers/admin/reports_controller.rb]: @ramanbedi1989 - actually i have to change this logic, becz there is no separate lop refund report.
It is combined with lop report.

File: app/controllers/admin/reports_controller.rb (hunk 1 above)
Comment (psahni) [inline app/controllers/admin/reports_controller.rb]: Yes you are right.

**Pull request merged**